docs/spring-server/http-request-response.md

```diff
@@ -0,0 +1,14 @@
+---
+id: http-request-response
+title: Access the HTTP Request-Response
+---
+
+Sometimes you may need access to the raw HTTP request and response object.
+There are multiple ways and locations you can access this information, however note that GraphQL has different paradigms than traditional REST APIs.
+Any data that is not part of the schema is no longer known by clients for free and it makes communication more difficult as this becomes an "undocumented" part of your API.
+
+That being said, there are common use cases like authentication and authorization that require inspecting HTTP headers.
+
+## GraphQL Context
+
+Using the [GraphQLContextFactory](./spring-graphql-context.md) you can access the request and response and add information into the GraphQL context on every operation which can then be used in your schema functions.
```

Comment:
maybe we should add a link to https://expediagroup.github.io/graphql-kotlin/docs/schema-generator/execution/contextual-data and then reference the spring implementation?

Reply:
If you are looking for the HTTP headers, you will be in the spring server library, so looking at the docs on how to generally use the `GraphQL Context` in the schema generator isn't that helpful. In the page I liked to we reference the schema generator docs.

Reply:
Lets rephrase it then to something like

```
The most common way to access the raw HTTP request and response objects is to process them when creating the `GraphQLContext` through [GraphQLContextFactory](./spring-graphql-context.md). Using the factory you can then extract the information from the incoming request and store it in the context so they can be accessed from any resolver.
```